src/prompt/cursor.rs

```diff
@@ -34,6 +55,24 @@ impl StringCursor {
         }
     }
 
+    pub fn move_word_left(&mut self) {
+        if self.cursor > 0 {
+            let jumps = word_jump_indices(&self.value);
+            let ix = jumps.binary_search(&self.cursor).unwrap_or_else(|i| i);
+            self.cursor = jumps[std::cmp::max(ix - 1, 0)];
+        }
+    }
+
+    pub fn move_word_right(&mut self) {
+        if self.cursor < self.value.len() {
+            let jumps = word_jump_indices(&self.value);
+            let ix = jumps
+                .binary_search(&self.cursor)
+                .map_or_else(|i| i, |i| i + 1);
+            self.cursor = jumps[std::cmp::min(ix, jumps.len() - 1)];
```

Comment:
Well, `jump.len()` will never be zero inside of this `if` condition, however, `saturating_sub` is just safer.
```suggestion
            self.cursor = jumps[std::cmp::min(ix, jumps.len().saturating_sub(1))];
```